src/servers/sapphire_api/PlayerMinimal.cpp

```diff
@@ -104,31 +108,58 @@ std::string PlayerMinimal::getModelString()
                             + std::to_string( m_modelEquip[ 1 ] ) + "\",\""
                             + std::to_string( m_modelEquip[ 2 ] ) + "\",\""
                             + std::to_string( m_modelEquip[ 3 ] ) + "\",\""
-                            + std::to_string( m_modelEquip[ 4 ] ) + "\",\"5\",\"6\",\"7\",\"8\",\"9\"";
+                            + std::to_string( m_modelEquip[ 4 ] ) + "\",\""
+                            + std::to_string( m_modelEquip[ 5 ] ) + "\",\""
+                            + std::to_string( m_modelEquip[ 6 ] ) + "\",\""
+                            + std::to_string( m_modelEquip[ 7 ] ) + "\",\""
+                            + std::to_string( m_modelEquip[ 8 ] ) + "\",\""
+                            + std::to_string( m_modelEquip[ 9 ] ) + "\"";
   return modelString;
 }
 
 std::string PlayerMinimal::getInfoJson()
 {
   std::string charDetails = "{\"content\":[\"" + std::string( getName() ) + "\"," +
-                            //"[" + getClassString() + "]," +
-                            "[\"0\",\"0\",\"0\",\"0\",\"0\",\"1\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\"],"
+                            "[\"0\",\"0\",\"0\",\"0\",\"" + std::to_string( getClassLevel( m_class ) ) +
+                            "\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\",\"0\"],"
                             "\"0\",\"0\",\"0\",\"" +
                             std::to_string( getBirthMonth() ) +
                             "\",\"" + std::to_string( getBirthDay() ) +
                             "\",\"" + std::to_string( getGuardianDeity() ) +
                             "\",\"" + std::to_string( m_class ) +
                             "\",\"0\",\"" + std::to_string( getZoneId() ) +
                             "\",\"0\"," +
-
                             "[" + getLookString() + "]," +
-                            "\"0\",\"0\"," +
+                            "\"" + std::to_string( m_modelMainWeapon ) + "\",\"" + std::to_string( m_modelSubWeapon ) + "\"," +
                             "[" + getModelString() + "]," +
-                            "\"1\",\"0\",\"0\",\"0\",\"0\",\"0\",\"\",\"0\",\"0\"]," +
+                            "\"1\",\"0\",\"0\",\"0\",\"" + std::to_string( m_equipDisplayFlags ) +
+                            "\",\"0\",\"\",\"0\",\"0\"]," +
                             "\"classname\":\"ClientSelectData\",\"classid\":116}";
   return charDetails;
 }
 
+uint16_t PlayerMinimal::getClassLevel(uint8_t classId)
```

Comment:
with a get function, you expect it to be just that, a getter. not something that sets object members.

Reply:
+ style